dockernet/ts-tests/test/utils.ts

```diff
@@ -0,0 +1,210 @@
+import { Coin, StdFee } from "@cosmjs/amino";
+import { fromBase64, toHex, toUtf8 } from "@cosmjs/encoding";
+import { ripemd160 } from "@noble/hashes/ripemd160";
+import { sha256 } from "@noble/hashes/sha256";
+import { bech32 } from "bech32";
+
+/**
+ * Creates a Coin object from the given string representation of a coin.
+ *
+ * @example
+ * ```
+ * coinFromString("1ustrd") => {amount:"1",denom:"ustrd"}
+ * ```
+ 
+* @param {string} coinAsString A string representation of a coin in the format "amountdenom"
+ * @returns {Coin} A Coin object with the extracted amount and denom
+ */
+export const coinFromString = (coinAsString: string): Coin => {
+  const regexMatch = coinAsString.match(/^([\d\.]+)([a-z]+)$/);
+
+  if (regexMatch === null) {
+    throw new Error(`cannot extract denom & amount from '${coinAsString}'`);
+  }
+
+  return { amount: regexMatch[1], denom: regexMatch[2] };
+};
+
+/**
+ * Converts a string of comma-separated coins into an array of Coin objects.
+ *
+ * @example
+ * ```
+ * coinsFromString("1ustrd,2uosmo") => => [{amount:"1",denom:"ustrd"},{amount:"2",denom:"uosmo"}]
+ * ```
+ *
+ * @param {string} coinsAsString A string of comma-separated coins in the format "amountdenom,amountdenom,..."
+ * @returns {Coin[]} An array of Coin objects
+ */
+export const coinsFromString = (coinsAsString: string): Coin[] =>
+  coinsAsString.split(",").map(coinFromString);
+
+/**
+ * Creates a StdFee object from the given gas limit and gas price.
+ *
+ * @param {number} gasLimit The gas limit to use for the fee calculation.
+ * @param {number} [gasPrice=0.025] The gas price to use for the fee calculation. Defaults to 0.025.
+ * @returns {StdFee} A StdFee object with the calculated fee amount and gas limit.
+ */
+export const feeFromGas = (
+  gasLimit: number,
+  gasPrice: number = 0.025,
+): StdFee => ({
+  amount: coinsFromString(`${gasLimit * gasPrice}ustrd`),
+  gas: String(gasLimit),
+});
+
+/**
+ * Compute the IBC denom of a token that was sent over IBC.
+ *
+ * @example
+ * To get the IBC denom of STRD on mainnet Osmosis:
+ * ```
+ * ibcDenom([{incomingPortId: "transfer", incomingChannelId: "channel-326"}], "ustrd")
+ * ```
+ *
+ * @param {Object[]} paths An array of objects containing information about the IBC transfer paths.
+ * @param {string} coinMinimalDenom The minimal denom of the coin.
+ * @returns {string} The computed IBC denom of the token.
+ */
+export const ibcDenom = (
```

Comment:
nit: I think this will always be the transfer channel so you can change the param to take just the channel ID and denom!

Reply:
with ibc contracts it's `wasm.{contract_address}`

Reply:
ah didn't realize that